### PR TITLE
Enable NXP USB Device controllers (udc_mcux) in Zephyr

### PR DESCRIPTION
**Pull request by MarkWangChinese**
Add the NXP USB Device controllers support in Zephyr.
Support EHCI and IP3511, verified on RT1060, LPC55S69 and RT685.

### CODE REVIEW

**Comment by dleach02** [comment]
@MarkWangChinese will the usb middleware changes be pulled back into an update to our source repository?

Also, please clean up the merge conflict.

**Review by dleach02 — APPROVED** [review]

**Comment by MarkWangChinese** [comment]
> @MarkWangChinese will the usb middleware changes be pulled back into an update to our source repository?
> 
> Also, please clean up the merge conflict.

Hi David, Do you mean https://github.com/nxp-mcuxpresso/mcux-sdk-middleware-usb or the SDK NXP internal repo? I will update the codes to SDK NXP internal repo and it will be released in next release.

File: mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_lpcip3511.c [hunk 1]
```diff
@@ -8,6 +8,11 @@
 
 #include "usb_device_config.h"
 #include "usb.h"
+/* CONFIG_UDC_DRIVER is for Zephyr, it will not be defined in NXP MCUXpresso SDK */
+#if ((defined CONFIG_UDC_DRIVER) && (CONFIG_UDC_DRIVER))
+#include "fsl_device_registers.h"
+#include "usb_device_mcux_drv_port.h"
+#else
```
Comment (mmahadevan108) [inline mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_lpcip3511.c:15]: Why is this change needed. We had no local changes to the SDK driver related to Zephyr earlier. All Zephyr related changes were kept inside the Zephyr driver.

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
We cannot have local changes inside the HAL that is maintained here. Please move all Zephyr related changes to the Zephyr driver.

File: mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_lpcip3511.c (hunk 1 above)
Comment (MarkWangChinese) [reply mcux/middleware/mcux-sdk-middleware-usb/device/usb_device_lpcip3511.c:15]: Hi Mahesh, I add more explanation in the README for it. The issue is: In order to use the SDK controller driver, Zephyr need to include many SDK device stack header files. I add one `usb_device_mcux_drv_port.h `to reduce the complexity/dependency to only need to include this header file. It makes `udc_mcux.c` to use SDK usb controller drivers more easily and more convenient/maintainable to adapter with Zephyr usb controller driver interface because many SDK structure/variables are not used in Zephyr environment.

Anyway, as I replied to David, I will put this change to SDK USB stack.

**Review by mmahadevan108 — APPROVED** [review]
